typing/typecore.mli

```diff
@@ -114,7 +114,7 @@ val type_expect:
 val type_exp:
         Env.t -> Parsetree.expression -> Typedtree.expression
 val type_approx:
-        Env.t -> Parsetree.expression -> type_expr
+        Env.t -> Parsetree.expression -> type_expr -> unit
```

Comment:
Yes it does. This is needed because you sometimes need the expected type to see that a function parameter is polymorphic. Without it `type_approx` guesses wrong causing spurious type errors.